client/src/components/LoginSession/tests/LoginSessionContainer-test.js

```diff
@@ -28,40 +39,153 @@ window.ss.config = {
     SecurityID: 1234567890
 };
 
-describe('LoginSessionContainer', () => {
-    let wrapper;
-    let props;
+const sessionData = {
+    IPAddress: '127.0.0.1',
+    IsCurrent: false,
+    UserAgent: 'Chrome on Mac OS X 10.15.7',
+    Created: '2021-01-20 00:33:41',
+    LastAccessed: '2021-03-11 03:47:22'
+};
+
+const props = {
+    // used in LoginSessionContainer
+    ID: 1,
+    LogOutEndpoint: '/test',
+    displayToastSuccess: jest.fn(() => 1),
+    displayToastFailure: jest.fn(() => 1),
+    // passed on to LoginSession
+    ...sessionData
+};
+
+MockDate.set('2021-03-12 03:47:22');
 
+describe('LoginSessionContainer', () => {
     beforeEach(() => {
-        MockDate.set('2021-03-12 03:47:22');
-        props = {
-            // used in LoginSessionContainer
-            ID: 1,
-            LogOutEndpoint: '/test',
-            displayToastSuccess: jest.fn(() => 1),
-            displayToastFailure: jest.fn(() => 1),
-            // passed on to LoginSession
-            IPAddress: '127.0.0.1',
-            IsCurrent: false,
-            UserAgent: 'Chrome on Mac OS X 10.15.7',
-            Created: '2021-01-20 00:33:41',
-            LastAccessed: '2021-03-11 03:47:22',
-        };
+        props.displayToastSuccess.mockClear();
+        props.displayToastFailure.mockClear();
     });
 
     it('should call displayToastSuccess on success', async () => {
         setEndpointFetcherSuccess(true);
-        wrapper = shallow(<LoginSessionContainer {...props} />);
-        await wrapper.instance().logout();
-        expect(props.displayToastSuccess).toBeCalledTimes(1);
-        expect(props.displayToastFailure).toBeCalledTimes(0);
+        let wrapper = shallow(<LoginSessionContainer {...props} />);
+        let loginSession = wrapper.find(LoginSession).first();
+
+        expect(loginSession.props()).toMatchObject({
```

Comment:
The test as initially written was only testing the end state. We can do better than that. We can test state transitions as the user interact with the component.

First, we test that `<LoginSession />` is rendered as expected in the initial state.